source/opt/local_redundancy_elimination.cpp

```diff
@@ -23,14 +23,10 @@ Pass::Status LocalRedundancyEliminationPass::Process(ir::IRContext* c) {
   InitializeProcessing(c);
 
   bool modified = false;
+  ValueNumberTable vnTable(context());
```

Comment:
If you are not resetting the VNs at every basic block, isn't this effectively doing global VN?

Reply:
No, the "value_to_ids" map gets reset for every basic block.  So the pass does not know which ids are available and contain the value number unless then come from the same basic block.

Reply:
Ah, ok.  So, this contains the global table, but it uses the local value_to_ids for the actual transformation.  Thanks.